### ISSUE TITLE
Properly handling randomness in the McICA algorithm of RRTMG within CLIMLAB

### ISSUE THREAD

**Issue by dorrih-stardust** [climlab/climlab#226]
Currently calls to RRTMG have no memory, meaning the seed is initialized within each call (If I understand it properly). That means that the cloud configuration is not really randomized, and I suspect this way the McICA is not properly executed.
We should probably consider passing information about the seed and number of taken time steps through the wrapper to RRTMG.

**Comment by dorrih-stardust** [comment]
I drilled a bit deeper into to topic of randomization in RRTMG, including in the CAM6 code. Some insights:

1. I now (somewhat) better understand what I previously hand-waived (literally) about regarding the sampled cloud configuration. The sampling involves a mapping between the frequency group number and the g-point. The function responsible for constructing cloud columns is called mcica_subcol_lw. It gets cloud parameters (cloud fraction, water and ice paths and particle radii) and optical parameters ($\tau$ for LW and $\tau,\omega$,g for SW) defined over the grid and the energy groups. It returns cloud fraction, water and ice paths and the optical parameters defined over the grid and the g-points. So I am not sure what it has to do with sub-grid sampling (it is actually randomized over the “spectral sub-grid”.

2. Inside the rrtmg code there is a function called generate_stochastic_clouds, which well documented, and was written by Cecile Hannay at UCAR based on Raisanen et al., QJRMS, 2004. Some insights:

(a) Seed: “If the stochastic cloud generator is called several times during the same timestep, one should change the seed between the call to insure that the subcolumns are different. This is done by changing the argument 'changeSeed' [DH: permuteseed in the calls to generate_stochastic_clouds]. For example, if one wants to create a set of columns for the shortwave and another set for the longwave, use 'changeSeed = 1' for the first call and 'changeSeed = 2' for the second call”. Strangely enough, neither in CLIMLAB nor in CAM6 do they change this parameter, meaning that we are basically sampling the same distribution in each column between calls.

(b) There are 3 options (originally 4, but the 'Exponential overlap' option was removed by Iacono) for handling cloud overlap: 1=random overlap, 2=max/random, 3=maximum overlap. The choice in both CAM6 (hardcoded) and CLIMLAB (default) is 1, but in principle one can also set it to 3 if one wishes to avoid randomness.

**The bottom line when it comes to randomness in cloud sampling: Both (CLIMLAB and CAM6) do not seem to be using RRTMG in the way it was intended in order to implement MC-ICA. Am I wrong in this interpretation?**

**Comment by dorrih-stardust** [comment]
Submitted PR #227 to resolve this issue
